Sources/NIOHTTP2/HTTP2ToHTTP1Codec.swift

```diff
@@ -613,6 +580,35 @@ internal extension HTTPResponseHead {
         let status = HTTPResponseStatus(statusCode: integerStatus)
         self.init(version: .init(major: 2, minor: 0), status: status, headers: HTTPHeaders(regularHeadersFrom: headers))
     }
+    
+    /// Create a `HTTPResponseHead` from the header block. Use this for non-informational responses
+    init(http2HeaderBlock headers: HPACKHeaders, requestHead: HTTPRequestHead, endStream: Bool) throws {
+        // A response head should have only one psuedo-header. We strip it off.
+        let statusHeader = try headers.peekPseudoHeader(name: ":status")
+        guard let integerStatus = Int(statusHeader, radix: 10) else {
+            throw NIOHTTP2Errors.invalidStatusValue(statusHeader)
+        }
+        let status = HTTPResponseStatus(statusCode: integerStatus)
```

Comment:
Given that the first few lines here are identical to the method above, should we not just call that and assign it to `self`, and then modify `self` as needed by adding the additional headers?